### DISCUSSION TITLE
Concurrent requests - "lock table"

### DISCUSSION THREAD

**Discussion by dominikjasek** [mikro-orm/mikro-orm#4069] [Q&A] [answered]
Hi,

let's consider following use case:

We are developing API for registering users on events. For example - event is `football match`, and can have maximally 10 users registered. We allow registration from 8pm.

At 8:00:01pm, 11 users try to register on this event using concurrent requests from frontend.

We don't want to allow registration to the last one. How to correctly handle this? Is there anything like `table lock`? Apparantely not. Is this database specific problem?

Thanks!

**Comment by B4nan** [comment, marked as answer]
You can lock a table via raw query, there is no native support for that, the `em.lock` is only for locking a single entity. I guess we could allow locking the full table this way too. You can always use a query builder, something like this could do the job:

```ts
await qb.select('1').setLockMode(LockMode.PESSIMISTIC_WRITE);
```

Alternatively, why not allow storing all the items, have an autoincrement column, and use just the first 10 items as valid? Depends on the use case I guess, another way would be using a queue. I had never solved this on a database level myself.
